### PR TITLE
DDCNL-1372: Making changes in Employment to take sequenceNumber in pl…

### PR DESCRIPTION
**Pull request by Sachin300793**
…ace of Key

- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

**Review by Christian-QA — APPROVED** [review]
LTGM

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala [hunk 1]
```diff
@@ -37,45 +37,46 @@ class AnnualAccountSpec extends PlaySpec {
...
     "Generate a full key (unique employer designation plus optional employee payroll number)" when {
       "Employment has an employee payrollNumber present" in {
 
-        val designation = SutWithNoPayments.key
-        designation mustBe "taxdistrict-payeref-payroll"
+        val designation = SutWithNoPayments.sequenceNumber
+        designation mustBe 0
       }
     }
     "Generate a full key consisting of only tax district and paye ref" when {
```
Comment (struong) [inline test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala:53]: rename the test please

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala [hunk 2]
```diff
@@ -37,45 +37,46 @@ class AnnualAccountSpec extends PlaySpec {
...
-      val designation = SutWithNoPayments.employerDesignation
-      designation mustBe "taxdistrict-payeref"
-    }
+    // TODO Can be removed
+//    "Generate a unique employer designation, consisting of tax district and paye ref extracted from the id string" in {
+//      val designation = SutWithNoPayments.employerDesignation
+//      designation mustBe "taxdistrict-payeref"
+//    }
 
     "Generate a full key (unique employer designation plus optional employee payroll number)" when {
```
Comment (struong) [inline test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala:46]: rename the test please

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala [hunk 3]
```diff
@@ -37,45 +37,46 @@ class AnnualAccountSpec extends PlaySpec {
       }
     }
 
-    "Generate a unique employer designation, consisting of tax district and paye ref extracted from the id string" in {
-      val designation = SutWithNoPayments.employerDesignation
-      designation mustBe "taxdistrict-payeref"
-    }
+    // TODO Can be removed
```
Comment (struong) [inline test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala:40]: Do the TODO

File: app/uk/gov/hmrc/tai/model/domain/AnnualAccount.scala [hunk 4]
```diff
@@ -20,19 +20,14 @@ import play.api.libs.json.{Format, Json}
 import uk.gov.hmrc.tai.model.TaxYear
 
 case class AnnualAccount(
-  key: String,
+  sequenceNumber: Int,
```
Comment (struong) [inline app/uk/gov/hmrc/tai/model/domain/AnnualAccount.scala:23]: Just wondering, do we even need this in the frontend?

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala (hunk 1 above)
Comment (Sachin300793) [reply test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala:53]: Done

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala (hunk 2 above)
Comment (Sachin300793) [reply test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala:46]: Done

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala (hunk 3 above)
Comment (Sachin300793) [reply test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala:40]: Done

File: app/uk/gov/hmrc/tai/model/domain/AnnualAccount.scala (hunk 4 above)
Comment (Sachin300793) [reply app/uk/gov/hmrc/tai/model/domain/AnnualAccount.scala:23]: I think we need it to match the Json coming from Tai backend

Comment (struong) [reply app/uk/gov/hmrc/tai/model/domain/AnnualAccount.scala:23]: Lets try removing it and see what happens

Comment (Sachin300793) [reply app/uk/gov/hmrc/tai/model/domain/AnnualAccount.scala]: Made this change. Its working as expected after removing. Done.

File: test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala [hunk 5]
```diff
@@ -37,25 +37,17 @@ class AnnualAccountSpec extends PlaySpec {
...
 
-    // TODO Can be removed
-//    "Generate a unique employer designation, consisting of tax district and paye ref extracted from the id string" in {
-//      val designation = SutWithNoPayments.employerDesignation
-//      designation mustBe "taxdistrict-payeref"
-//    }
-
-    "Generate a full key (unique employer designation plus optional employee payroll number)" when {
+    "Check sequenceNumber" when {
       "Employment has an employee payrollNumber present" in {
```
[the remark this answers is not in the document]
Comment (Sachin300793) [reply test/uk/gov/hmrc/tai/model/domain/AnnualAccountSpec.scala]: Already removed this test , not sure why its not reflecting here

**Review by struong — APPROVED** [review]

**Pull request merged**